src/Symfony/Component/Security/Http/Event/SwitchUserEvent.php

```diff
@@ -24,11 +25,13 @@ class SwitchUserEvent extends Event
 {
     private $request;
     private $targetUser;
+    private $token;
 
-    public function __construct(Request $request, UserInterface $targetUser)
+    public function __construct(Request $request, UserInterface $targetUser, TokenInterface $token)
```

Comment:
adding a new required argument is a BC break. Instead, you should either make it optional, or rely solely on the setter